zebrad/tests/acceptance.rs

```diff
@@ -1473,3 +1487,413 @@ where
+    zebra_directory: TempDir,
+    timeout: Duration,
+) -> Result<(TestChild<TempDir>, SocketAddr)> {
+    let mut config = random_known_rpc_port_config()
+        .expect("Failed to create a config file with a known RPC listener port");
+
+    config.state.ephemeral = false;
+    config.network.initial_mainnet_peers = HashSet::new();
+    config.network.initial_testnet_peers = HashSet::new();
+    config.mempool.debug_enable_at_height = Some(0);
+
+    let mut zebrad = zebra_directory
+        .with_config(&mut config)?
+        .spawn_child(&["start"])?
+        .with_timeout(timeout)
+        .with_failure_regex_iter(
+            // TODO: replace with a function that returns the full list and correct return type
+            ZEBRA_FAILURE_MESSAGES
+                .iter()
+                .chain(PROCESS_FAILURE_MESSAGES)
+                .cloned(),
+            NO_MATCHES_REGEX_ITER.iter().cloned(),
+        );
+
+    let rpc_address = config.rpc.listen_addr.unwrap();
+
+    zebrad.expect_stdout_line_matches("activating mempool")?;
+    zebrad.expect_stdout_line_matches(&format!("Opened RPC endpoint at {}", rpc_address))?;
+
+    Ok((zebrad, rpc_address))
+}
+
+/// Start a lightwalletd instance with its RPC server functionality enabled.
+///
+/// Returns the lightwalletd instance and the port number that it is listening for RPC connections.
+fn spawn_lightwalletd_with_rpc_server(
+    zebrad_rpc_address: SocketAddr,
+) -> Result<(TestChild<TempDir>, u16)> {
+    let lightwalletd_dir = testdir()?.with_lightwalletd_config(zebrad_rpc_address)?;
+
+    let lightwalletd_rpc_port = random_known_port();
+    let lightwalletd_rpc_address = format!("127.0.0.1:{lightwalletd_rpc_port}");
+
+    let mut arguments = vec!["--grpc-bind-addr".to_owned(), lightwalletd_rpc_address];
+
+    if let Ok(data_dir) = env::var(LIGHTWALLETD_DATA_DIR_VAR) {
+        arguments.push("--data-dir".to_owned());
+        arguments.push(data_dir);
+    }
+
+    let borrowed_arguments: Vec<&str> = arguments.iter().map(|string| string.as_ref()).collect();
+    let mut lightwalletd = lightwalletd_dir
+        .spawn_lightwalletd_child(&borrowed_arguments)?
+        .with_timeout(LIGHTWALLETD_TEST_TIMEOUT)
+        .with_failure_regex_iter(
+            // TODO: replace with a function that returns the full list and correct return type
+            LIGHTWALLETD_FAILURE_MESSAGES
+                .iter()
+                .chain(PROCESS_FAILURE_MESSAGES)
+                .cloned(),
+            // TODO: some exceptions do not apply to the cached state tests (#3511)
+            LIGHTWALLETD_IGNORE_MESSAGES.iter().cloned(),
+        );
+
+    lightwalletd.expect_stdout_line_matches("Starting gRPC server")?;
+    lightwalletd.expect_stdout_line_matches("Waiting for block")?;
+
+    Ok((lightwalletd, lightwalletd_rpc_port))
+}
+
+/// Connect to a lightwalletd RPC instance.
+async fn connect_to_lightwalletd(lightwalletd_rpc_port: u16) -> Result<LightwalletdRpcClient> {
```

Comment:
Optional: We could move this utility function to one of the test sub-modules.

Reply:
Done.